docs/reference/api.md

```diff
@@ -0,0 +1,77 @@
+# API
+
+Pebble exposes an API to allow remote clients to interact with the daemon. The API can start and stop services, add configuration layers to the plan, and so on.
+
+## Accessing the API
+
+The API uses HTTP over a Unix socket, with access to the API controlled by user ID. If `pebble run` is started with the `--http <address>` option, Pebble exposes a limited set of open-access API endpoints using the given TCP address. See {ref}`api-access-levels`.
+
+See below for some examples of how to use the API. For more examples, see "How to use the Pebble API". <!-- [David] Link to the how-to guide -->
+
+There's a Go client for the API. See the [Go client documentation](https://pkg.go.dev/github.com/canonical/pebble/client) and the examples below.
+
+We try to never change the underlying HTTP API in a backwards-incompatible way. However, in rare cases we may change the Go client in a backwards-incompatible way.
+
+There's also a Python client for the API. The Python client is part of the `ops` library used by Juju charms. See the [Ops documentation](https://ops.readthedocs.io/en/latest/).
+
+### Go client examples
+
+TODO: go client code snippets.
+
+### cURL examples
+
+TODO: curl examples.
+
+## API access levels
+
+API endpoints fall into one of three access levels, from least restricted to most restricted. See {ref}`api-access-levels`.
+
+### Identities
+
+Pebble has the concept of named "identities", which lets administrators manage users' access to the API. See {ref}`controlling-api-access-using-identities`.
+
+## Common parameters
+
+### timeout
+
+The format of the timeout is a decimal number with an optional unit suffix (e.g., "300ms", "1.5s", "2h45m").
+
+Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+
+## Common errors
```

Comment:
Looking over the rendered docs again, I think we should actually move all error codes to this section, and move them all out of each endpoint. They don't add value there.

So I think this section should be just "Errors" and have an introductory paragraph like so:

> API endpoints may return errors as 4xx status codes (client errors) and 5xx status codes (server errors). Errors are always returned in JSON format with the following structure:
>
>     { ... }

And then just a bulleted list or table of error codes:

* 400: bad request, for example, a badly-formatted parameter
* 404: not found, for example, a change with the specified ID does not exist
* ...
* 500: internal server error, for example, the Pebble database is corrupt; if these errors keep happening, please [open an issue](link)